core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeExternalSorter.java

```diff
@@ -191,24 +191,28 @@ public void spill() throws IOException {
       spillWriters.size(),
       spillWriters.size() > 1 ? " times" : " time");
```

Comment:
Not sure whether we should move this log statement inside the `if` block or not.  I suppose it might be useful to know when memory pressure triggered a spill even if we didn't end up writing rows, so it's probably fine to leave this where it is.